core/src/mast/node/mod.rs

```diff
@@ -143,6 +143,40 @@ impl MastNode {
         }
     }
 
+    /// Remap the Node to its new position using the given [`Remapping`].
+    pub fn remap(&mut self, remapping: &Remapping) {
```

Comment:
Let's explain briefly in the doc comments what the remapping actually does (e.g., we recursively update all the children of this node in such and such way).